--- src/Makefile.am~	2006-07-20 06:21:36.000000000 +0200
+++ src/Makefile.am	2009-01-30 22:05:17.000000000 +0100
@@ -25,7 +25,8 @@
 	-avoid-version \
 	-rpath $(moduledir) \
 	-module \
-	$(LIBTOOL_EXPORT_OPTIONS) \
+	$(LIBTOOL_EXPORT_OPTIONS)
+thai_la_LIBADD = \
 	$(SCIM_LIBS) \
 	$(LIBTHAI_LIBS)
 
@@ -42,7 +43,8 @@
 	-avoid-version \
 	-rpath $(moduledir) \
 	-module \
-	$(LIBTOOL_EXPORT_OPTIONS) \
+	$(LIBTOOL_EXPORT_OPTIONS)
+thai_imengine_setup_la_LIBADD = \
 	$(SCIM_GTKUTILS_LIBS) \
 	$(SCIM_LIBS)
 endif
--- src/Makefile.in~	2008-05-24 09:02:01.000000000 +0200
+++ src/Makefile.in	2009-01-30 22:06:35.000000000 +0100
@@ -256,7 +256,8 @@
 	-avoid-version \
 	-rpath $(moduledir) \
 	-module \
-	$(LIBTOOL_EXPORT_OPTIONS) \
+	$(LIBTOOL_EXPORT_OPTIONS)
+thai_la_LIBADD = \
 	$(SCIM_LIBS) \
 	$(LIBTHAI_LIBS)
 
@@ -269,7 +270,8 @@
 @SCIM_BUILD_SETUP_TRUE@	-avoid-version \
 @SCIM_BUILD_SETUP_TRUE@	-rpath $(moduledir) \
 @SCIM_BUILD_SETUP_TRUE@	-module \
-@SCIM_BUILD_SETUP_TRUE@	$(LIBTOOL_EXPORT_OPTIONS) \
+@SCIM_BUILD_SETUP_TRUE@	$(LIBTOOL_EXPORT_OPTIONS)
+@SCIM_BUILD_SETUP_TRUE@thai_imengine_setup_la_LIBADD = \
 @SCIM_BUILD_SETUP_TRUE@	$(SCIM_GTKUTILS_LIBS) \
 @SCIM_BUILD_SETUP_TRUE@	$(SCIM_LIBS)
 
